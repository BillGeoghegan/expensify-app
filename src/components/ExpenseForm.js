import React from 'react';
// Moment for date object
import moment from 'moment';
// Date picker and styles
import { SingleDatePicker } from 'react-dates';


export default class ExpenseForm extends React.Component {
    constructor(props){
        super(props);

        this.state = {
            description: props.expense ? props.expense.description : '',
            note: props.expense ? props.expense.note : '',
            amount: props.expense ? (props.expense.amount/100).toString() : '',
            createdAt: props.expense ? moment(props.expense.createdAt) : moment(),
            calendarFocused: false,
            error: undefined
        };
    }
    onDescriptionChange = (e) => {
        const description = e.target.value;
        this.setState(() => ({ description }));
    };
    onNoteChange = (e) => {
        const note = e.target.value;    //must set it to variable as it's a live state and constantly changing...
        this.setState(() => ({ note }))
    };
    onAmountChange = (e) => {
        const amount = e.target.value;
        // Regex:
        // If starts with (^) a digit (d) as between 1 and infinity ({1,})
        // And COULD ('(\.\d{0,2})?') include a '.' followed by a digit that is between 0 and 2 characters long
        // And ends after that ($)
        if(!amount || amount.match(/^\d{1,}(\.\d{0,2})?$/)) {
            this.setState( () => ({amount}));
        }
    };
    onDateChange = (createdAt) => {
        if(createdAt){
            this.setState( () => ({createdAt}));
        }        
    };
    onFocusChange = ({focused}) => {
        this.setState( () => ({calendarFocused: focused}));
    };
    onSubmit = (e) => {
        e.preventDefault();
        if(!this.state.description || !this.state.amount){
            // Set error state equal to 'Please provide description and amount.'
            this.setState( () => ({error: 'Please provide description and amount.'}));
        } else {
            // Clear the error
            this.setState( () => ({error: undefined}));
            this.props.onSubmit({
                description: this.state.description,
                amount: parseFloat(this.state.amount,10) * 100, //parse from string 
                createdAt: this.state.createdAt.valueOf(),      //get moment value
                note: this.state.note
            })
            console.log('Submitted.');
        }
    };
    render() {
        return (
            <div>
                {this.state.error && <p>{this.state.error}</p>}
                <form onSubmit={this.onSubmit}>
                    <input
                        type="text"
                        placeholder="Description"
                        autoFocus
                        value={this.state.description}
                        onChange={this.onDescriptionChange}
                    />
                    <input
                        type="number"
                        placeholder="Amount"
                        value={this.state.amount}
                        onChange={this.onAmountChange}
                    />
                    <SingleDatePicker 
                        date={this.state.createdAt}
                        onDateChange={this.onDateChange}
                        focused={this.state.calendarFocused}
                        onFocusChange={this.onFocusChange}
                        numberOfMonths={1}              //Show only one month picker instead of 2 default
                        isOutsideRange={() => false}    //Make every day false without any logic test
                    />
                    <textarea
                        placeholder="Add a note for your expense (optional)"
                        value={this.state.note}
                        onChange={this.onNoteChange}
                    >
                    </textarea>
                    <button
                        type="submit"
                        value="Submit"
                    >Add Expense</button>
                </form>
            </div>
        )
    }
}