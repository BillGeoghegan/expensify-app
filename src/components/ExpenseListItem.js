import React from 'react';
import { Link } from 'react-router-dom';

const ExpenseListItem = ({dispatch, id, description = '', note = '', amount = 0, createdAt = 0}) => {
    return (
        <div>
            <Link to={`/edit/${id}`}>
                <h3>{description}</h3>
            </Link>
            <p>{amount} - {createdAt}</p>
            <p>{note}</p>
        </div>
    );
}

export default ExpenseListItem;