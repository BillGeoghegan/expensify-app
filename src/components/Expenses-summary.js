import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total';

export const ExpensesSummary = ({expenseCount, expensesTotal}) => {
    const expenseWord = expenseCount === 1 ? 'expense' : 'expenses'
    const formattedExpenseTotal = numeral(expensesTotal/100).format('$0,0.00');
    if(expenseCount === 0){
        return(
            <div>
                <h1>Currently not viewing anything.</h1>
            </div>
        );
    }else{
        return(
            <div>
                <h1>Viewing {expenseCount} {expenseWord} totalling {formattedExpenseTotal}</h1>
            </div>
        );
    }
};

const MapStateToProps = (state) => {
    const visibleExpenses = selectExpenses(state.expenses, state.filters);
    return{
        expenseCount: visibleExpenses.length,
        expensesTotal: selectExpensesTotal(visibleExpenses)
    };
};

export default connect(MapStateToProps)(ExpensesSummary);