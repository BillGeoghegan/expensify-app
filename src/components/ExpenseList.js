import React from 'react';
import { connect } from 'react-redux';
import ExpenseListItem from './ExpenseListItem';
import selectExpenses from '../selectors/expenses';

//List of props to return
export const ExpenseList = (props) => (
    <div>
        <h1>Expense List</h1>
        {props.expenses.map((expense) => {
            return <ExpenseListItem key={expense.id} {...expense}/>;
        })}
    </div>
);

//Using Redux to map out all expenses
const mapStateToProps = (state) => {
    return {
        expenses: selectExpenses(state.expenses, state.filters)
    };
};

//Using Redux fetched data to pass back the now populated list by connecting the two functions
export default connect(mapStateToProps)(ExpenseList);