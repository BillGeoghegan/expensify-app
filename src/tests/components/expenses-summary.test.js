import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/expenses-summary';

test('Should return 0 value if empty array', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={0} expensesTotal={235} />);
    expect(wrapper).toMatchSnapshot();
});

test('Should summarize all values if passed an array', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={3} expensesTotal={24535} />);
    expect(wrapper).toMatchSnapshot();
});

test('Should summarize a single value if passed an array', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={235} />);
    expect(wrapper).toMatchSnapshot();
});
