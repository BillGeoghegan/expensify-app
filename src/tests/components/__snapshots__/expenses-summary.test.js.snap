// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should return 0 value if empty array 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ExpensesSummary
    expenseCount={0}
    expensesTotal={235}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="content-container"
>
        <h1
                className="page-header__title"
        >
                Viewing 
                <span />
                 
                expenses
                 totalling 
                <span>
                        $2.35
                </span>
        </h1>
        <div
                className="page-header__actions"
        >
                <Link
                        className="button"
                        replace={false}
                        to="/Create"
                >
                        Add Expense
                </Link>
        </div>
</div>,
      "className": "page-header",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="page-header__title"
>
            Viewing 
            <span />
             
            expenses
             totalling 
            <span>
                        $2.35
            </span>
</h1>,
          <div
            className="page-header__actions"
>
            <Link
                        className="button"
                        replace={false}
                        to="/Create"
            >
                        Add Expense
            </Link>
</div>,
        ],
        "className": "content-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Viewing ",
              <span />,
              " ",
              "expenses",
              " totalling ",
              <span>
                $2.35
</span>,
            ],
            "className": "page-header__title",
          },
          "ref": null,
          "rendered": Array [
            "Viewing ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": 0,
              },
              "ref": null,
              "rendered": 0,
              "type": "span",
            },
            " ",
            "expenses",
            " totalling ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "$2.35",
              },
              "ref": null,
              "rendered": "$2.35",
              "type": "span",
            },
          ],
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Link
              className="button"
              replace={false}
              to="/Create"
>
              Add Expense
</Link>,
            "className": "page-header__actions",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Add Expense",
              "className": "button",
              "replace": false,
              "to": "/Create",
            },
            "ref": null,
            "rendered": "Add Expense",
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="content-container"
>
          <h1
                    className="page-header__title"
          >
                    Viewing 
                    <span />
                     
                    expenses
                     totalling 
                    <span>
                              $2.35
                    </span>
          </h1>
          <div
                    className="page-header__actions"
          >
                    <Link
                              className="button"
                              replace={false}
                              to="/Create"
                    >
                              Add Expense
                    </Link>
          </div>
</div>,
        "className": "page-header",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1
              className="page-header__title"
>
              Viewing 
              <span />
               
              expenses
               totalling 
              <span>
                            $2.35
              </span>
</h1>,
            <div
              className="page-header__actions"
>
              <Link
                            className="button"
                            replace={false}
                            to="/Create"
              >
                            Add Expense
              </Link>
</div>,
          ],
          "className": "content-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Viewing ",
                <span />,
                " ",
                "expenses",
                " totalling ",
                <span>
                  $2.35
</span>,
              ],
              "className": "page-header__title",
            },
            "ref": null,
            "rendered": Array [
              "Viewing ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": 0,
                },
                "ref": null,
                "rendered": 0,
                "type": "span",
              },
              " ",
              "expenses",
              " totalling ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "$2.35",
                },
                "ref": null,
                "rendered": "$2.35",
                "type": "span",
              },
            ],
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Link
                className="button"
                replace={false}
                to="/Create"
>
                Add Expense
</Link>,
              "className": "page-header__actions",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Add Expense",
                "className": "button",
                "replace": false,
                "to": "/Create",
              },
              "ref": null,
              "rendered": "Add Expense",
              "type": [Function],
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Should summarize a single value if passed an array 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ExpensesSummary
    expenseCount={1}
    expensesTotal={235}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="content-container"
>
        <h1
                className="page-header__title"
        >
                Viewing 
                <span>
                        1
                </span>
                 
                expense
                 totalling 
                <span>
                        $2.35
                </span>
        </h1>
        <div
                className="page-header__actions"
        >
                <Link
                        className="button"
                        replace={false}
                        to="/Create"
                >
                        Add Expense
                </Link>
        </div>
</div>,
      "className": "page-header",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="page-header__title"
>
            Viewing 
            <span>
                        1
            </span>
             
            expense
             totalling 
            <span>
                        $2.35
            </span>
</h1>,
          <div
            className="page-header__actions"
>
            <Link
                        className="button"
                        replace={false}
                        to="/Create"
            >
                        Add Expense
            </Link>
</div>,
        ],
        "className": "content-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Viewing ",
              <span>
                1
</span>,
              " ",
              "expense",
              " totalling ",
              <span>
                $2.35
</span>,
            ],
            "className": "page-header__title",
          },
          "ref": null,
          "rendered": Array [
            "Viewing ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": 1,
              },
              "ref": null,
              "rendered": 1,
              "type": "span",
            },
            " ",
            "expense",
            " totalling ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "$2.35",
              },
              "ref": null,
              "rendered": "$2.35",
              "type": "span",
            },
          ],
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Link
              className="button"
              replace={false}
              to="/Create"
>
              Add Expense
</Link>,
            "className": "page-header__actions",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Add Expense",
              "className": "button",
              "replace": false,
              "to": "/Create",
            },
            "ref": null,
            "rendered": "Add Expense",
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="content-container"
>
          <h1
                    className="page-header__title"
          >
                    Viewing 
                    <span>
                              1
                    </span>
                     
                    expense
                     totalling 
                    <span>
                              $2.35
                    </span>
          </h1>
          <div
                    className="page-header__actions"
          >
                    <Link
                              className="button"
                              replace={false}
                              to="/Create"
                    >
                              Add Expense
                    </Link>
          </div>
</div>,
        "className": "page-header",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1
              className="page-header__title"
>
              Viewing 
              <span>
                            1
              </span>
               
              expense
               totalling 
              <span>
                            $2.35
              </span>
</h1>,
            <div
              className="page-header__actions"
>
              <Link
                            className="button"
                            replace={false}
                            to="/Create"
              >
                            Add Expense
              </Link>
</div>,
          ],
          "className": "content-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Viewing ",
                <span>
                  1
</span>,
                " ",
                "expense",
                " totalling ",
                <span>
                  $2.35
</span>,
              ],
              "className": "page-header__title",
            },
            "ref": null,
            "rendered": Array [
              "Viewing ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": 1,
                },
                "ref": null,
                "rendered": 1,
                "type": "span",
              },
              " ",
              "expense",
              " totalling ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "$2.35",
                },
                "ref": null,
                "rendered": "$2.35",
                "type": "span",
              },
            ],
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Link
                className="button"
                replace={false}
                to="/Create"
>
                Add Expense
</Link>,
              "className": "page-header__actions",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Add Expense",
                "className": "button",
                "replace": false,
                "to": "/Create",
              },
              "ref": null,
              "rendered": "Add Expense",
              "type": [Function],
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Should summarize all values if passed an array 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ExpensesSummary
    expenseCount={3}
    expensesTotal={24535}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="content-container"
>
        <h1
                className="page-header__title"
        >
                Viewing 
                <span>
                        3
                </span>
                 
                expenses
                 totalling 
                <span>
                        $245.35
                </span>
        </h1>
        <div
                className="page-header__actions"
        >
                <Link
                        className="button"
                        replace={false}
                        to="/Create"
                >
                        Add Expense
                </Link>
        </div>
</div>,
      "className": "page-header",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="page-header__title"
>
            Viewing 
            <span>
                        3
            </span>
             
            expenses
             totalling 
            <span>
                        $245.35
            </span>
</h1>,
          <div
            className="page-header__actions"
>
            <Link
                        className="button"
                        replace={false}
                        to="/Create"
            >
                        Add Expense
            </Link>
</div>,
        ],
        "className": "content-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Viewing ",
              <span>
                3
</span>,
              " ",
              "expenses",
              " totalling ",
              <span>
                $245.35
</span>,
            ],
            "className": "page-header__title",
          },
          "ref": null,
          "rendered": Array [
            "Viewing ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": 3,
              },
              "ref": null,
              "rendered": 3,
              "type": "span",
            },
            " ",
            "expenses",
            " totalling ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "$245.35",
              },
              "ref": null,
              "rendered": "$245.35",
              "type": "span",
            },
          ],
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Link
              className="button"
              replace={false}
              to="/Create"
>
              Add Expense
</Link>,
            "className": "page-header__actions",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Add Expense",
              "className": "button",
              "replace": false,
              "to": "/Create",
            },
            "ref": null,
            "rendered": "Add Expense",
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="content-container"
>
          <h1
                    className="page-header__title"
          >
                    Viewing 
                    <span>
                              3
                    </span>
                     
                    expenses
                     totalling 
                    <span>
                              $245.35
                    </span>
          </h1>
          <div
                    className="page-header__actions"
          >
                    <Link
                              className="button"
                              replace={false}
                              to="/Create"
                    >
                              Add Expense
                    </Link>
          </div>
</div>,
        "className": "page-header",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1
              className="page-header__title"
>
              Viewing 
              <span>
                            3
              </span>
               
              expenses
               totalling 
              <span>
                            $245.35
              </span>
</h1>,
            <div
              className="page-header__actions"
>
              <Link
                            className="button"
                            replace={false}
                            to="/Create"
              >
                            Add Expense
              </Link>
</div>,
          ],
          "className": "content-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Viewing ",
                <span>
                  3
</span>,
                " ",
                "expenses",
                " totalling ",
                <span>
                  $245.35
</span>,
              ],
              "className": "page-header__title",
            },
            "ref": null,
            "rendered": Array [
              "Viewing ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": 3,
                },
                "ref": null,
                "rendered": 3,
                "type": "span",
              },
              " ",
              "expenses",
              " totalling ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "$245.35",
                },
                "ref": null,
                "rendered": "$245.35",
                "type": "span",
              },
            ],
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <Link
                className="button"
                replace={false}
                to="/Create"
>
                Add Expense
</Link>,
              "className": "page-header__actions",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Add Expense",
                "className": "button",
                "replace": false,
                "to": "/Create",
              },
              "ref": null,
              "rendered": "Add Expense",
              "type": [Function],
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
